https://issues.redhat.com/browse/ACM-19151--2.13.2 additional content needed now

swopebe opened the pull request.

swopebe committed:
https://issues.redhat.com/browse/ACM-19151

swopebe committed:
Update known_issues_install.adoc

jacobbaungard commented on release_notes/known_issues_install.adoc:
```diff
@@ -69,7 +71,7 @@ oc delete crd multiclusterobservabilities.observability.open-cluster-management.
 - If the Observability component is `enabled` on the cluster, the previous command deletes the entire Observability stack. 
 +
-- No known workaround exists for an upgrade with Observability enabled. You can unistall {acm-short} completely and reinstall without any CRDs without this failure.
+- No known workaround exists for an upgrade with Observability enabled. You can unistall {acm-short} completely and reinstall without any CRDs without this failure, or you can upgrade to 2.13.2. 
```
With 2.13.2 out now, I believe the workaround is to start an upgrade to 2.13.2. As I understand it even if you started an upgrade to 2.13.1, and got stuck due to this, when an upgrade to 2.13.2 is started afterwards the error should clear. That is probably the workaround we should recommend in all scenarios first now. @dislbenn what do you think?

swopebe replied:
@jacobbaungard I have another PR to tell users just to upgrade to 2.13.2.

That being said, I need to know from both you and the install team if we need to remove this release note as a result. Or remove all of the process in the release note. I didn't get an issue to remove all that, but I can under this issue if that is the change.

jacobbaungard replied:
I would recommend that we keep the release note in general.

What I would like to communicate to customers now with 2.13.2 out, is that if they already updated to 2.13.1, and are seeing the issue, they should upgrade to 2.13.2, even if their system is at a state where their update to 2.13.1 is "stuck". I am not sure there is much reason to recommend the original workaround when 2.13.2 is out, as the original workaround has some caveats.

I don't see the other PR you refer to.

dislbenn replied:
@swopebe I agree with @jacobbaungard for this case. I think it's important to mention if they have updated to 2.13.1, they can upgrade to 2.13.2 as the official workaround to resolve this issue.

swopebe replied:
I need to know what to do with this content:

`. With the Observability component `disabled` on your `MultiClusterHub` custom resource instance, run the following command to delete the Observabiliity `crd`:

+
[source,bash]
----
oc delete crd multiclusterobservabilities.observability.open-cluster-management.io
----
+
*Important:* 
+
- If the Observability component is `enabled` on the cluster, the previous command deletes the entire Observability stack. 
+
- No known workaround exists for an upgrade with Observability enabled. You can unistall {acm-short} completely and reinstall without any CRDs without this failure, or you can upgrade to 2.13.2. 
+
- If you do not want to lose the Observability stack, do not updrade {acm-short} with {ocp-short} 4.18.

. After you delete the Observability CRD, run the following command to check the progress of the {acm-short} upgrade:

+
[source,bash]
----
oc get multiclusterhubs -n open-cluster-management -o yaml
----

+
The output states that hub cluster components are ready, as in the following example:

+
[source,bash]
----
message: 'created new resource: CustomResourceDefinition multiclusterobservabilities.observability.open-cluster-management.io'
 reason: NewResourceCreated
 status: "True"
 type: Progressing
 message: All hub components ready.
 ...
 reason: ComponentsAvailable
 status: "True"
 type: Complete
----`

swopebe replied:
I guess I was not clear in my question--sorry about that. I am adding that they need to upgrade to 2.13.2.  Seems like we agree line 40 - 103 can be removed, or do we need to keep the verification portion of that?

swopebe replied:
I have made some edits, but I didn't remove everything, yet.

swopebe committed:
Update known_issues_install.adoc

swopebe commented on release_notes/known_issues_install.adoc:
```diff
@@ -31,8 +31,12 @@ For more about deprecations and removals, see xref:../release_notes/acm_deprecat
+To resolve this error, upgrade to {acm-short} 2.13.2.
+
+*Note:* If you upgrade directly to 2.13.2, you don't receive this error. If you install {acm-short} 2.13 with {ocp-short} 4.18 for the first time, you do not receive this error. 
+// please look at line 40 - 103 -- does all that go away now?
```
See comment in line 39--TY.

swopebe commented on release_notes/known_issues_install.adoc:
```diff
@@ -31,8 +31,12 @@ For more about deprecations and removals, see xref:../release_notes/acm_deprecat
-If you upgrade an existing {acm-short} 2.13 cluster with {ocp-short} 4.18, your upgrade fails with an error from installing a CRD. If you install {acm-short} 2.13 with {ocp-short} 4.18 for the first time, you do not receive this error.
+If you upgrade an existing {acm-short} 2.13 cluster with {ocp-short} 4.18, your upgrade fails with an error from installing a CRD. 
 
+To resolve this error, upgrade to {acm-short} 2.13.2.
```
new line

swopebe replied:
resolved in a meeting, see commit description.

swopebe committed:
Update known_issues_install.adoc

swopebe committed:
Update known_issues_install.adoc

swopebe committed:
Update known_issues_install.adoc

swopebe committed:
adding a general statement for upgrade

swopebe commented on release_notes/known_issues_install.adoc:
```diff
@@ -27,19 +27,34 @@ For your {ocp} cluster, see link:https://docs.redhat.com/documentation/en-us/ope
 
 For more about deprecations and removals, see xref:../release_notes/acm_deprecate_remove.adoc#deprecations-removals-acm[Deprecations and removals for {acm-short}].
 
+[#upgrade-stuck]
```
This is the general upgrade note that To Hung asked me for so my question here is do we need both? If so, it's fine. Just let me know.

swopebe committed:
Update known_issues_install.adoc

After meeting with Install lead, we decided to combine the general statement to upgrade to 2.13.2 with the observability details, remove the work around, and test to get the initial status for the user.

swopebe committed:
Update known_issues_install.adoc

dislbenn commented on release_notes/known_issues_install.adoc:
```diff
@@ -27,13 +27,15 @@ For your {ocp} cluster, see link:https://docs.redhat.com/documentation/en-us/ope
+When you upgrade to {acm-short} 2.13 with an {ocp-short} version 4.18 cluster, you receive an error installing the CRD and remain in a `stuck` status. 
 
-. To get the details of the error, run the following command:
+To resolve this issue, upgrade directly to {acm-short} 2.13.2, even if you if you upgraded to 2.13.1 and receive this error.
```
```suggestion
To resolve this issue, upgrade directly to {acm-short} 2.13.2, even if you upgraded to 2.13.1 and receive this error.
```

dislbenn commented on release_notes/known_issues_install.adoc:
```diff
@@ -57,30 +59,14 @@ desiredVersion: 2.13.0
-- If you do not want to lose the Observability stack, do not updrade {acm-short} with {ocp-short} 4.18.
-
-. After you delete the Observability CRD, run the following command to check the progress of the {acm-short} upgrade:
+Run the following command to check the progress of the {acm-short} upgrade:
```
I think this is okay to leave in there, since it’s showing the customer that they can use the command to validate the MCH CR status after the workaround has been applied.

jacobbaungard commented on release_notes/known_issues_install.adoc:
```diff
@@ -27,13 +27,15 @@ For your {ocp} cluster, see link:https://docs.redhat.com/documentation/en-us/ope
-[#observability-upgrade-issue]
-== Upgrading an existing {acm-short} cluster with {ocp-short} 4.18 creates an error installing a CRD
+[#upgrade-stuck]
+== Upgrading an existing {acm-short} cluster with {ocp-short} 4.18 creates an error stuck in `upgrading` status while installing the CRD 
```
```suggestion
== Upgrading an existing {acm-short} cluster with {ocp-short} 4.18 causes the upgrade to get stuck in `upgrading` status while installing the CRD 
```

The sentence sounded a bit weird - around the "an error stuck in". Attempted to rewrite slightly.

swopebe committed:
Update release_notes/known_issues_install.adoc

Co-authored-by: Disaiah Bennett <dbennett@redhat.com>

swopebe committed:
Update release_notes/known_issues_install.adoc

Co-authored-by: Jacob Baungård Hansen <jacobbaungard@redhat.com>

jacobbaungard reviewed (approved):
/lgtm

swopebe committed:
Update known_issues_install.adoc

jc-berger commented on release_notes/known_issues_install.adoc:
```diff
@@ -27,13 +27,15 @@ For your {ocp} cluster, see link:https://docs.redhat.com/documentation/en-us/ope
-[#observability-upgrade-issue]
-== Upgrading an existing {acm-short} cluster with {ocp-short} 4.18 creates an error installing a CRD
+[#upgrade-stuck]
+== Upgrading an existing {acm-short} cluster with {ocp-short} 4.18 causes the upgrade to get stuck in `upgrading` status while installing the CRD 
```
I thought we were trying to make short titles? 

What do you think of this: 

```suggestion
== Upgrading an existing {acm-short} cluster with {ocp-short} 4.18 

Upgrading an existing {acm-short} cluster with {ocp-short} 4.18 causes the upgrade to get stuck in `upgrading` status while installing the CRD. 
```

jc-berger replied:
Or if you want it to be more about the user fixing, maybe this option as a header title:

```suggestion
== Fixing a stuck _upgrading_ status 

Upgrading an existing {acm-short} cluster with {ocp-short} 4.18 causes the upgrade to get stuck in `upgrading` status while installing the CRD 
```

swopebe replied:
We have to be more specific here because it is an ACM hub with **4.18** OCP. A general upgrade known issue can be any supported version of OCP as a hub cluster, this is specific to 4.18. 

We need to discuss the problems a bit more in the known issue title to help support quickly scan based on the user's feedback. Before I created the title, I asked what the user initially sees without contacting support or digging any further, because that is what they will show support.

During upgrade with OCP 4.18 as the ACM hub, they get **to** the process of installing a CRD and it becomes stuck in "upgrading."

jc-berger reviewed (approved):
/lgtm
